public/js/save.js

```diff
@@ -124,26 +170,34 @@ $(document).ready(function() {
         counter++;
     }
     //Push All Data
-    function pushData() { //TODO: Check for connectivity before running.
+    function pushData(alertUser) {
         var error = false;
-        for (var i = (counter-1); i >= 1; i--) {
-            var data = JSON.parse(get('form', i));
-            var request = $.ajax({
-                type: 'POST',
-                url: '/pushData',
-                data: data
-            });
-            request.done(function(response) { //If pushing is successful.
-                console.log("Data successfully pushed.");
-            });
-            request.fail(function(jqXHR, textStatus) { //If pushing is unsuccessful.
-                if(!error){
-                    alert("Error while pushing. Please reach out to a qualified individual for assistance. Error Message: " + textStatus);
-                    error = true;
-                }
-            });
+        var dataArray = [];
+        for (var i = 0; i < counter; i++) {
```

Comment:
You should be using `let` inside for-loops. Using var like this means that it is accessible from the entire parent scope.